### PR TITLE
Configure Qodana to check code quality

### PR DESCRIPTION
**Pull request by rob93c**
Along with Qodana configuration, this pull request adds the project icon to be used by IntelliJ.

### CODE REVIEW

**Comment by rob93c** [comment]
@MartelliEnrico the changes are now ready to be reviewed

**Comment by MartelliEnrico** [comment]
It looks like it still doens't know about Java 20 Preview: https://qodana.cloud/projects/3XQOZ/reports/2WqN6

**Comment by rob93c** [comment]
> It looks like it still doens't know about Java 20 Preview: https://qodana.cloud/projects/3XQOZ/reports/2WqN6

The issue has been reported to Qodana's team and one of their maintainers is looking into this: https://github.com/JetBrains/Qodana/discussions/181#discussioncomment-6630488

**Comment by rob93c** [comment]
Qodana team identified an issue preventing it from enabling preview features, tracked in [QD-6990](https://youtrack.jetbrains.com/issue/QD-6990/Support-using-Java-preview-features).

The ticket can therefore be merged now, we will ignore the 4 alerts for the time being @MartelliEnrico

File: .idea/icon.svg
Comment (MartelliEnrico) [inline .idea/icon.svg:1]: Are we sure we want this inside this PR?

Comment (rob93c) [reply .idea/icon.svg:1]: Yes ✨

**Pull request merged**